src/wasm/wasm-emscripten.cpp

```diff
@@ -404,7 +406,32 @@ struct EmJsWalker : public PostWalker<EmJsWalker> {
     int64_t address = addrConst->value.getInteger();
     auto code = stringTracker.codeForConstAddr(address);
     codeByName[funcName] = code;
+    codeAddresses[address] = code.size() + 1;
+  }
+};
+
+struct SegmentRemover : WalkerPass<PostWalker<SegmentRemover>> {
+  SegmentRemover(Index segment) : segment(segment) {}
+
+  bool isFunctionParallel() override { return true; }
+
+  Pass* create() override { return new SegmentRemover(segment); }
+
+  void visitMemoryInit(MemoryInit* curr) {
+    if (segment == curr->segment) {
+      Builder builder(*getModule());
+      replaceCurrent(builder.makeNop());
+    }
```

Comment:
This would be more robust against side effects if it created a block that evaluated and dropped all the operands of the `MemoryInit`, but perhaps that's overkill considering how much we can assume about the output of wasm-ld here.

Reply:
I see... I wonder if there are other places this is done?  Maybe there is some kind of helper for this already?

Reply:
I don't know of a general utility for this, but here's an example where this is done: https://github.com/WebAssembly/binaryen/blob/74ccdb87b3390e003b658b847526dad8722083f8/src/passes/MemoryPacking.cpp#L413-L416

`Builder::blockify` makes this simple enough that we probably don't need a separate utility for it (although a dedicated utility couldn't hurt).